tonic/src/codec/decode.rs

```diff
@@ -38,12 +38,43 @@ impl<T> Unpin for Streaming<T> {}
 
 #[derive(Debug, Clone)]
 enum State {
-    ReadHeader,
+    ReadHeader {
+        span: Option<tracing::Span>,
+    },
     ReadBody {
+        span: tracing::Span,
         compression: Option<CompressionEncoding>,
         len: usize,
     },
-    Error(Status),
+    Error(Box<Status>),
```

Comment:
Boxing here significantly reduces the size of `State` and `StreamingInner` as a result, even with the addition of `tracing::Span` to the other variants.